java-checks-test-sources/src/main/java/checks/TooManyParametersCheck.java

```diff
@@ -103,3 +119,96 @@ public void foo4(String p1, String p2, String p3, String p4, String p5, String p
+  public SpringRepository(String p1, String p2, String p3, String p4, String p5, String p6, String p7, String p8) {} // Compliant
+  
+  public void foo5(String p1, String p2, String p3, String p4, String p5, String p6, String p7, String p8) {} // Noncompliant
+}
+
+@Repository
+class SpringRepositoryMoreThanOneConstructor{
+  public SpringRepositoryMoreThanOneConstructor(String p1, String p2, String p3, String p4, String p5, String p6, String p7, String p8) {} // Noncompliant
+  public SpringRepositoryMoreThanOneConstructor(String p1, String p2, String p3) {}; // Compliant
+  
+  public void foo5(String p1, String p2, String p3, String p4, String p5, String p6, String p7, String p8) {} // Noncompliant
+}
+
+@Service
+class SpringService{
+  public SpringService(String p1, String p2, String p3, String p4, String p5, String p6, String p7, String p8) {} // Compliant
+  
+  public void foo5(String p1, String p2, String p3, String p4, String p5, String p6, String p7, String p8) {} // Noncompliant
+}
+
+@Service
+class SpringServiceMoreThanOneConstructor{
+  public SpringServiceMoreThanOneConstructor(String p1, String p2, String p3, String p4, String p5, String p6, String p7, String p8) {} // Noncompliant
+  public SpringServiceMoreThanOneConstructor(String p1, String p2, String p3) {}; // Compliant
+  
+  public void foo5(String p1, String p2, String p3, String p4, String p5, String p6, String p7, String p8) {} // Noncompliant
+}
+
+class MicronautHttpAnnotations{
+  
+  @Get
+  public void get(String p1, String p2, String p3, String p4, String p5, String p6, String p7, String p8) {} // Compliant
+  @Post
+  public void post(String p1, String p2, String p3, String p4, String p5, String p6, String p7, String p8) {} // Compliant
+  @Put
+  public void put(String p1, String p2, String p3, String p4, String p5, String p6, String p7, String p8) {} // Compliant
+  @Delete
+  public void delete(String p1, String p2, String p3, String p4, String p5, String p6, String p7, String p8) {} // Compliant
+  @Options
+  public void options(String p1, String p2, String p3, String p4, String p5, String p6, String p7, String p8) {} // Compliant
+  @Patch
+  public void patch(String p1, String p2, String p3, String p4, String p5, String p6, String p7, String p8) {} // Compliant
+  @Head
+  public void head(String p1, String p2, String p3, String p4, String p5, String p6, String p7, String p8) {} // Compliant
+  @Trace
+  public void trace(String p1, String p2, String p3, String p4, String p5, String p6, String p7, String p8) {} // Compliant
+  
+}
+
+/* 
+ * We can maybe have a discussion on records' constructors, if they should be checked against the rule,
+ * but since they are used to store articulate data, it might not be a good idea
+ * The one note to keep in mind is that it seems like a record could be used as a Spring @Component
+ */
+record Record1(String p1, String p2, String p3, String p4, String p5, String p6, String p7, String p8) {} // Compliant 
```

Comment:
Once we understand this rule, it's better to take a decision about records now and not trigger a discussion for later.
If someone comes back to this rule, it may be in 2 years, perhaps not us, and it will cost some effort to understand the rule again.
The two of us are enough to make a decision.
I vote to have a specific exclusion about records because records are data classes where all the fields are always set through the constructor parameters, it's the only solution. And I have the same point of view for annotations. So I would add those exceptions:
```java
  public List<Tree.Kind> nodesToVisit() {
    // This rule has the following exceptions: RECORD, ANNOTATION_TYPE
    return Arrays.asList(Tree.Kind.CLASS, Tree.Kind.INTERFACE, Tree.Kind.ENUM);
  }
``` 
And if we make an exception about records, we should update the RSPEC, and just having one compliant test for records is enough, but it should also be with an additional constructor, not only the compact constructor, here it's cheating.
But if we don't exclude records, as a general rule, then it's better to always have two kinds of tests: `Compliant` and `Noncompliant`.
Same for enum, interface, and annotation, if we decide to raise issues on them, we should have `Noncompliant` tests.

Reply:
Yes I agree that we should ignore them as exceptions to the rule. I will add compliant tests for both records and annotations

Reply:
I think annotations cannot really have parameters in their methods, nor they can have a constructor. Am I right? If that is the case I can skip tests for them I think